### PR TITLE
feat: add auth&metric&traverser&task api and ut

### PR DESCRIPTION
**Pull request by simon824**
add ut & pyhugegraph code format

### CODE REVIEW

File: .github/workflows/hugegraph-python-client.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
         pip install -r ./hugegraph-python-client/requirements.txt
     - name: Prepare HugeGraph Server Environment
       run: |
-        docker run -d --name=graph -p 8080:8080 hugegraph/hugegraph
+        docker run -d --name=graph -p 8080:8080 -e AUTH=true -e PASSWORD=admin hugegraph/hugegraph:1.2.0
```
Comment (imbajin) [inline .github/workflows/hugegraph-python-client.yml:29]: no need to set `-e AUTH-true` now (the latest PR https://github.com/apache/incubator-hugegraph/pull/2427/files enhance it)

Note: the docker-1.2.0 image need update first

**Comment by simon824** [comment]
@imbajin @lzyxx77 @liuxiaocs7 @javeme  please take a review when free, thanks!

File: hugegraph-python-client/src/pyhugegraph/api/auth.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+
+class AuthManager(HugeParamsBase):
+    def __init__(self, graph_instance):
+        super().__init__(graph_instance)
+        self.session = self.set_session(HugeSession.new_session())
+
+    def set_session(self, session):
+        self.session = session
+        return session
```
Comment (liuxiaocs7) [inline hugegraph-python-client/src/pyhugegraph/api/auth.py:34]: Is it possible to simplify here? eg: `self.__session = HugeSession.new_session()`, same below

File: hugegraph-python-client/src/pyhugegraph/api/schema_manage/property_key.py [hunk 3]
```diff
@@ -160,6 +170,7 @@ def eliminate(self):
         )
         if check_if_success(response, error):
             return f"eliminate PropertyKey success, Detail: {str(response.content)}"
+        return f"eliminate PropertyKey failed, Detail: {response.content}"
```
Comment (liuxiaocs7) [inline hugegraph-python-client/src/pyhugegraph/api/schema_manage/property_key.py:173]: need `str(response.content)` when fail?

File: hugegraph-python-client/src/pyhugegraph/client.py [hunk 4]
```diff
@@ -63,3 +70,27 @@ def variable(self):
             return self._variable
         self._variable = VariableManager(self._graph_instance)
         return self._variable
+
+    def auth(self):
+        if self._auth:
+            return self._auth
+        self._auth = AuthManager(self._graph_instance)
+        return self._auth
```
Comment (liuxiaocs7) [inline hugegraph-python-client/src/pyhugegraph/client.py:78]: can be simplify like below?

```python
self._auth = self._auth or AuthManager(self._graph_instance)
return self._auth
```

same below

**Review by liuxiaocs7 — COMMENTED** [review]
I noticed that there are two forms of test code in UT, as shown below, should they be standardized into one?

```
assert 4 == len(schema)

self.assertEqual(len(all_basic_metrics), 5)
```

**Review by javeme — APPROVED** [review]
+1

**Review by liuxiaocs7 — APPROVED** [review]

**Pull request merged**